diff -ruN ultrastardx.orig/dists/ultrastardx.desktop ultrastardx/dists/ultrastardx.desktop
--- ultrastardx.orig/dists/ultrastardx.desktop	2009-01-18 22:08:40.452008495 +0100
+++ ultrastardx/dists/ultrastardx.desktop	2009-01-18 22:27:15.409001549 +0100
@@ -3,6 +3,8 @@
 Version=1.0
 
 Name=UltraStar Deluxe
+GenericName=Karaoke game
+GenericName[de]=Karaoke-Spiel
 Comment=Karaoke program that evaluates your performance
 Comment[de]=Singe Karaoke und messe dich mit anderen Spielern
 
@@ -14,4 +16,4 @@
 Terminal=false
 
 Type=Application
-Categories=Application;Game;ArcadeGame;
+Categories=Game;ArcadeGame;
