diff -ruN ultrastardx.orig/game/fonts/fonts.ini ultrastardx/game/fonts/fonts.ini
--- ultrastardx.orig/game/fonts/fonts.ini	2009-12-21 21:38:54.560540217 +0100
+++ ultrastardx/game/fonts/fonts.ini	2009-12-21 21:39:14.862665570 +0100
@@ -1,11 +1,11 @@
 [Normal]
-File=FreeSans/FreeSans.ttf
+File=/usr/share/fonts/liberation/LiberationSans-Regular.ttf
 
 [Bold]
-File=FreeSans/FreeSansBold.ttf
+File=/usr/share/fonts/liberation/LiberationSans-Bold.ttf
 
 [Outline1]
-File=DejaVu/DejaVuSans-Bold.ttf
+File=/usr/share/fonts/liberation/LiberationSans-Bold.ttf
 
 [Outline2]
-File=FreeSans/FreeSansBold.ttf
+File=/usr/share/fonts/liberation/LiberationSans-Bold.ttf
